### DISCUSSION TITLE
Flashing Sonoff 3.0 Version E

### DISCUSSION THREAD

**Discussion by apla2016** [Koenkk/Z-Stack-firmware#497] [General]
How to flash Sonoff 3.0 Version E? I use to flash regular Sonoff with SmartRF Flash Programmer with no problem, but same procedure does not work with -E. Could it be that -E is built on other that CC2652P chip?

**Comment by tube0013** [comment]
Yes, the -E is using a Silicon Labs EFR32 MG21 radio.

Try: https://darkxst.github.io/silabs-firmware-builder/

**Comment by apla2016** [comment]
Thank you for super quick response. I tried this silabs online flasher, but it requires firmware in different format, not HEX that you developped.

Anatoly

**Comment by apla2016** [comment]
May be I did not get what you are saying. Are you saying to flash one of their firmware, or use their flasher to load your firmware?

**Reply by tube0013** [reply]
This repo is not mine, it is also firmware only for TI CC based radios. The site I linked has firmware for the sonoff-E.

**Comment by Hedda** [comment]
Sonoff ZBDongle-E (a.k.a. Dongle-E / ZBDongleE) by ITead indeed uses Silabs EFR32MG21 SoC and not radio chip from Texas Instruments.

Only ITead's Sonoff ZBDongle-P (a.k.a. Dongle-P / ZBDongleP) is based on the CC2652P radio chip from Texas Instruments.

To clearify; recommend use these unofficial community firmware builds for the Sonoff ZBDongle-E (a.k.a. Dongle-E / ZBDongleE) by ITead:

https://github.com/darkxst/silabs-firmware-builder/tree/main/firmware_builds/zbdonglee

For more information about using later Silicon Labs EmberZNet firmware with Zigbee2MQTT, see -> https://github.com/Koenkk/zigbee2mqtt/discussions/21462

Note that ITead/Sonoff themselves have not released any newer official firmware updates since that product was made available:

https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/NCP

Regardless,, it is specifically recommend to use a EmberZNet NCP firmware with 115200 speed for Z2M (not a RCP based firmware).

FYI, flashing of EmberZNet (EZSP / Ember) based adapters can also be done with thís Universal Silicon Labs Flasher (universal-silabs-flasher):

https://github.com/NabuCasa/universal-silabs-flasher

If you are using Zigbee2MQTT in Home Assistant then a tip is that there are several add-ons that are based on that for easier flashing:

https://github.com/darkxst/multipan_flasher/blob/main/README.md

https://github.com/tube0013/tubeszb_addons/blob/main/tzb-silabs-flasher/DOCS.md

https://github.com/mercenaruss/zigstar_addons/blob/main/zigstar-silabs-flasher/README.md

PS: Note that you first need to stop the Zigbee2MQTT application/add-on before you can flash the adapter.
